### PR TITLE
[DataGridPro] Keep bottom pinned row at the bottom

### LINKED ISSUE #12925: [data grid] The bottom pinned row in a data grid does not stay at the bottom when there aren't any other rows in the data set
### Steps to reproduce

Link to live example (7.3.1): https://stackblitz.com/edit/react-yp5dyf-mtr1cq?file=Demo.tsx

This appears to have broken with version 7.0.0.
6.19.11: https://stackblitz.com/edit/react-yp5dyf?file=Demo.tsx
7.0.0: https://stackblitz.com/edit/react-yp5dyf-1hxyj2?file=Demo.tsx

### Current behavior

The bottom pinned row appears below the header

### Expected behavior

The bottom pinned row should stay at the bottom of the grid

### Context

I'm trying to display a grid of data with a pinned row at the bottom with the totals for each of the columns

### Your environment

_No response_

**Search keywords**: pinned pin

### PR DESCRIPTION
**Pull request by romgrk**
Closes #12925

Before: https://codesandbox.io/p/sandbox/funny-leaf-6fv4qt
After: https://codesandbox.io/p/sandbox/mui-mui-x-x-data-grid-forked-9vpwwk

### CODE REVIEW

**Review by KenanYusuf — APPROVED** [review]

File: packages/x-data-grid/src/components/virtualization/GridVirtualScroller.tsx
Comment (MBilalShafi) [inline packages/x-data-grid/src/components/virtualization/GridVirtualScroller.tsx:1]: When [having](https://codesandbox.io/p/sandbox/mui-mui-x-x-data-grid-forked-t2k74k?file=%2Fsrc%2Fdemo.tsx%3A35%2C25) top and bottom pinned rows, it seems to introduce an unwanted scroll.


https://github.com/mui/mui-x/assets/12609561/ada69efc-33c4-45be-a349-4b47d3cdf233

Is this a separate/known issue?

Comment (romgrk) [reply packages/x-data-grid/src/components/virtualization/GridVirtualScroller.tsx:1]: This exists on master but I'll look into it.

Comment (romgrk) [reply packages/x-data-grid/src/components/virtualization/GridVirtualScroller.tsx:1]: It exists on master and is a separate issue but I've included a fix for it in this PR.

**Review by cherniavskii — COMMENTED** [review]
Looks good!
Can you add a unit test for this regression?

**Review by cherniavskii — APPROVED** [review]

File: packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx [hunk 1]
```diff
@@ -513,22 +513,13 @@ export const useGridVirtualScroller = () => {
   );
 
   const contentSize = React.useMemo(() => {
-    // In cases where the columns exceed the available width,
-    // the horizontal scrollbar should be shown even when there're no rows.
-    // Keeping 1px as minimum height ensures that the scrollbar will visible if necessary.
-    const height = Math.max(contentHeight, 1);
-
```
Comment (romgrk) [inline packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx:520]: I think this code was preventing [this 0-height warning](https://github.com/mui/mui-x/blob/5ef1694752a5fe50fac5a74f6835ec7fe0779c9f/packages/x-data-grid/src/hooks/features/dimensions/useGridDimensions.ts#L334-L345) from running, so fixing it broke some tests all over the place that didn't have explicit dimensions on the grid container. I'll need to debug & update those tests before merging this PR. The test runner also doesn't seem to match the warning to which test it originated from precisely, so sometimes it marks one test as failed but the warning came from another test. Lots of fun ahead.

**Pull request merged**